src/components/popover/widgets/capture-keystroke/capture-keystroke-button.component.ts

```diff
@@ -1,23 +1,77 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, HostListener, Output } from '@angular/core';
+import { CaptureService } from '../../../../services/capture.service';
 
 @Component({
     selector: 'capture-keystroke-button',
     template: require('./capture-keystroke-button.component.html'),
     styles: [require('./capture-keystroke-button.component.scss')]
 })
-export class CaptureKeystrokeButtonComponent implements OnInit {
+export class CaptureKeystrokeButtonComponent {
+    @Output() capture = new EventEmitter<any>();
+
     private record: boolean;
+    private first: boolean;
+
+    constructor(private captureService: CaptureService) {
+        this.record = false;
+        this.captureService.initModifiers();
+        this.captureService.populateMapping();
+    }
 
-    constructor() { }
+    @HostListener('keyup')
+    onKeyUp() {
+        if (this.record && !this.first) {
+            this.saveScanCode();
+        }
+    }
+
+    @HostListener('keydown', ['$event'])
+    onKeyDown(e: KeyboardEvent) {
+        const code: number = e.keyCode;
 
-    ngOnInit() { }
+        if (this.record) {
+            this.first = false;
+
+            if (this.captureService.hasMap(code)) {
+                this.saveScanCode(this.captureService.getMap(code));
+            } else {
+                this.captureService.setModifier((e.location === 1), code);
+            }
+        } else if (code === 13) {
+            this.record = true;
+            this.first = true;
+        }
+    }
+
+    @HostListener('focusout')
+    onFocusOut() {
+        this.record = false;
+        this.reset();
+    }
 
     start(): void {
-        this.record = true;
+        if (!this.record) {
```

Comment:
But it doesn't matter the capturing has already started or not. At the end of this function `record` will be true, no matter what is its value at the beginning.  (If it was true we don't touch it, if it was false we set it to true. It is the same is set to true, no matter the initial value.)